firmware/Makefile

```diff
@@ -10,6 +13,20 @@ build: FORCE
 clean: FORCE
 	rm $(BUILD_DIR) -rf
 
+deepclean: clean
+	@echo "Deleting files and directories in $(CUBEMX_DIR) except $(KEEP_FILES)"
+	@cd $(CUBEMX_DIR) && for file in *; do \
+		if ! echo "$(KEEP_FILES)" | grep -q "\\b$$file\\b"; then \
```

Comment:
if you use `git check-ignore`, then you will get a list of all files which should be deleted. Just call `rm <all those files>` instead of looping